src/docker/composebuilder/service_builder.go

```diff
@@ -24,9 +24,17 @@ type DevelopmentDockerComposeBuilder struct {
 	ServiceEndpoints         map[string]*ServiceEndpoints
 }
 
-// NewDevelopmentDockerComposeBuilder is DevelopmentDockerComposeBuilder's constructor
-func NewDevelopmentDockerComposeBuilder(appConfig types.AppConfig, serviceConfig types.ServiceConfig, serviceData types.ServiceData, role, appDir string, mode BuildMode, serviceEndpoints map[string]*ServiceEndpoints) *DevelopmentDockerComposeBuilder {
-	return &DevelopmentDockerComposeBuilder{
+// GetServiceDockerConfig returns the DockerConfigs for a service and its dependencies in docker-compose.yml
+func GetServiceDockerConfig(appConfig types.AppConfig, serviceConfig types.ServiceConfig, serviceData types.ServiceData, role string, appDir string, mode BuildMode, serviceEndpoints map[string]*ServiceEndpoints) (types.DockerConfigs, error) {
+	return NewServiceComposeBuilder(appConfig, serviceConfig, serviceData, role, appDir, mode, serviceEndpoints).getServiceDockerConfigs()
+}
+
+// NewServiceComposeBuilder is ServiceComposeBuilder's constructor
+func NewServiceComposeBuilder(appConfig types.AppConfig, serviceConfig types.ServiceConfig, serviceData types.ServiceData, role, appDir string, mode BuildMode, serviceEndpoints map[string]*ServiceEndpoints) *ServiceComposeBuilder {
+	if mode.Environment == BuildModeEnvironmentTest {
+		role = appConfig.GetTestRole(role)
+	}
+	return &ServiceComposeBuilder{
```

Comment:
New code encapsulates what was done in the now deleted `composebuilder/service.go`